pkg/kudoctl/packages/verifier/template/verify_render_test.go

```diff
@@ -62,3 +63,49 @@ spec:
 		`decoding chunk "\napiVersion: batch/v1\nkind: Job\nmetadata:\n  name: backup\nspec:\n  template:\n    spec:\n      containers:\n        - name: backup\n`+
 			`       restartPolicy: Never\n" failed: error converting YAML to JSON: yaml: line 10: did not find expected key`, res.Errors[0])
 }
+
+func TestTemplateRenderVerifierParameterTypes(t *testing.T) {
+	params := []v1beta1.Parameter{
+		v1beta1.Parameter{
+			Name:    "labels",
+			Default: kudo.String("a: a\nb: b"),
```

Comment:
Now I'm really confused. In the design phase you [agreed](https://github.com/kudobuilder/kudo/pull/1356#discussion_r383805749) that this should *not* be a string. Yet in the merged KEP has the exact opposite: [*everything* is a string](https://github.com/kudobuilder/kudo/blob/master/keps/0024-parameter-enhancement.md#support-optional-parameter-types), even the list which was not initially.

I don't think this is a good idea from consistency PoV: defaults for parameters should be defined in the same way as their values in a parameter value files. Otherwise we'll end up with a strange mix of YAML and YAML-shaped strings.
We should instead change the type of `Default` to be `interface{}`, like we agreed during the design.

Reply:
I agree, having the Default as actual yaml would be nicer.

Reply:
I just noticed an "issue" with this: The parameter values are always provided as strings (even if we're going to have the param-value-files from KEP-26 they're going to be serialized when installing/updating the instance).

If we're going to provide the defaults as yaml, we have to decide if we want to serialize these values somewhere, or have a bit more complex code when actually using the value or the default. I think having the actual yaml in the operatorversion would still make sense.

Reply:
This isn't possible with Kubernetes objects. `interface{}` isn't allowed by `controller-gen`. Hence, all values have to be provided as strings.

Reply:
Hm, really? IIRC k8s supports [oneOf](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) so it should be possible to make it work? :thinking:

Reply:
It's `protobuf` under the hood, which has [any](https://developers.google.com/protocol-buffers/docs/proto3#any), but it's a problem with `controller-gen` not supporting this. I'm currently looking at `controller-gen` docs and tickets to see if it is somehow possible to support it. Agree with you that actual YAML would be much nicer than YAML in strings.

Reply:
[runtime.RawExtension](https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/types.go#L94) or `runtime.Unknown` could get us there. Will run a few tests.

Reply:
There isn't support for untyped values in `controller-gen`. Types like `runtime.RawExtension` or `runtime.Unknown` still require API objects. Therefore, we have to use strings here.